### PR TITLE
[RISCV] Fix InsnCI register type

### LINKED ISSUE #100112: [RISCV] Incorrect definition of InsnCI
According to the spec:

> 26.2. Compressed Instruction Formats
> CR, CI, and CSS can use any of the 32 RVI registers, but CIW, CL, CS, CA, and CB are limited to just 8 of them.

But the definition in `RISCVInstrInfoC.td` has it taking `AnyRegC`. This should be `AnyReg`.

```
def InsnCI : DirectiveInsnCI<(outs AnyRegC:$rd), (ins uimm2_opcode:$opcode,
```

### PR DESCRIPTION
**Pull request by svs-quic**
According to the spec the CI type instructions can take any of the 32 RVI registers.

Fixes #100112

### CODE REVIEW

**Review by lenary — APPROVED** [review]
LGTM. Nice catch. Please wait for another reviewer before merging though.

**Review by wangpc-pp — APPROVED** [review]
LGTM.

**Comment by svs-quic** [comment]
Could someone please merge this? Thanks.

**Comment by svs-quic** [comment]
Does this need to go into the 19.x release branch?

**Pull request merged**
